juju/juju
# feat(apiserver): thin out Machiner api by replacing Jobs with IsController

This is a part of thinning out the `Machiner` api. 

- `Jobs()` is replaced by the new `IsController()` function backed by the machine domain service on dqlite. Machine service is wired up through the servicefactory in facade register.
- `stateconverter` worker is changed to use the new `IsController` function (instead of `Jobs()`).
- `Machiner` facade is revved to version 6. The new version doesnt't have the `Jobs` function, and has the new `IsController` function. The old version (5) has the `Jobs()` and the `SetMachineAddresses()` as stubs. The `SetMachineAddresses` is currently kept as it is, and it is to be replaced later by (and possibly moved to) the network api when link layer devices is added.

## Checklist




## QA steps

```
 $ juju bootstrap localhost deleteme && juju add-model test
 $ juju model-config "logging-config=juju.worker.stateconverter=TRACE"
 $ juju deploy ubuntu
```

In the debug log, make sure you see:

`machine-0: 02:31:43 TRACE juju.worker.stateconverter Calling Handle for machine-0`

## Links



**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/

**Jira card:** JUJU-6247
